gui/wxpython/datacatalog/tree.py

```diff
@@ -893,6 +910,50 @@ def OnSwitchLocationMapset(self, event):
             self.changeLocation.emit(mapset=self.selected_mapset[0].label, location=self.selected_location[0].label)
         self.ExpandCurrentMapset()
 
+    def OnCreateMapset(self, event):
+        """Create new mapset"""
+        self.gisdbase = gisenv()['GISDBASE']
+        self.location = gisenv()['LOCATION_NAME']
+
+        dlg = NewMapsetDialog(
+            parent=self,
+            default=get_default_mapset_name(),
+            database=self.gisdbase,
+            location=self.location
+        )
+        if dlg.ShowModal() == wx.ID_OK:
+            mapset = dlg.GetValue()
+            try:
+                create_mapset(self.gisdbase, self.location, mapset)
+                self.InsertMapset(name=mapset, location_node=self.location,
```

Comment:
location_node is a of type class DictNode (treemodel.py), not string. Why to pass twice mapset?

Reply:
Not sure how to simplify InsertMapset function. And not sure what parameter name means. I think it is necessary to include it but in this case it does not make much sense.